enumerables.rb

```diff
@@ -0,0 +1,220 @@
+      end
+    elsif is_a? Array
+      arr_to_work.my_each do |item|
+        return false unless yield(item)
+      end
+    elsif is_a? Hash
+      arr_to_work.my_each do |k, v|
+        return false unless yield(k, v)
+      end
+    end
+    true
+  end
+
+  def my_any?(pattern = nil) # rubocop:disable Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity
+    arr_to_work = self
+    if !block_given? && pattern.nil?
+      arr_to_work.my_each do |item|
+        return true if item
+      end
+    elsif pattern.is_a? Regexp
+      arr_to_work.my_each do |item|
+        return true if item =~ pattern
+      end
+    elsif pattern.is_a? Class
+      arr_to_work.my_each do |item|
+        return true if item.is_a? pattern
+      end
+    elsif pattern
+      arr_to_work.my_each do |item|
+        return true if item == pattern
+      end
+    elsif is_a? Array
+      arr_to_work.my_each do |item|
+        return true if yield(item)
+      end
+    elsif is_a? Hash
+      arr_to_work.my_each do |k, v|
+        return true if yield(k, v)
+      end
+    end
+    false
+  end
+
+  def my_none?(pattern = nil) # rubocop:disable Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity
+    arr_to_work = self
+    if !block_given? && pattern.nil?
+      arr_to_work.my_each do |item|
+        return false if item
+      end
+    elsif pattern.is_a? Regexp
+      arr_to_work.my_each do |item|
+        return false if item =~ pattern
+      end
+    elsif pattern.is_a? Class
+      arr_to_work.my_each do |item|
+        return false if item.is_a? pattern
+      end
+    elsif pattern
+      arr_to_work.my_each do |item|
+        return false if item == pattern
+      end
+    elsif is_a? Array
+      arr_to_work.my_each do |item|
+        return true unless yield(item)
+      end
+    elsif is_a? Hash
+      arr_to_work.my_each do |k, v|
+        return false if yield(k, v)
+      end
+    end
+    true
+  end
+
+  def my_count(xxx = nil) # rubocop:disable Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity
+    return length unless block_given? || !xxx.nil?
+
+    arr_to_work = self
+    counter = 0
+    if !xxx.nil?
+      arr_to_work.my_each do |item|
+        counter += 1 if item == xxx
+      end
+    elsif is_a? Array
+      arr_to_work.my_each do |item|
+        counter += 1 if yield(item)
+      end
+    elsif is_a? Hash
+      arr_to_work.my_each do |k, v|
+        counter += 1 if yield(k, v)
+      end
+    end
+    counter
+  end
+
+  def my_map
+    return to_enum unless block_given?
+
+    arr_to_work = self
+    new_array = []
+    if is_a? Array
+      arr_to_work.my_each do |k|
+        new_array.push(yield(k))
+      end
+    elsif is_a? Hash
+      arr_to_work.my_each do |k, v|
+        new_array.push(yield(k, v))
+      end
+    end
+    new_array
+  end
+
+  def my_inject(*args) # rubocop:disable Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity
+    my_arr = to_a
+    if block_given?
+      my_arr = to_a
```

Comment:
[REQUIRED]
Let's change `my_arr = to_a`  to `my_arr = dup.to_a`. Since we've used `shift` method which mutates the original array. Creating a duplicate will solve the problem. 
Let's try as much to avoid changing the array on which the method is called on.
This will fix this error :point_down:  
![inj](https://user-images.githubusercontent.com/44072711/68614238-87e4b000-048e-11ea-8090-2d2bff6c6411.png)

Reply:
thanks for the tip :)